website_chd_product_configurator/controllers/controllers.py

```diff
@@ -0,0 +1,237 @@
+                message = 'image uploaded successfully'
+            except:
+                message = 'Problems uploading your image, please contact us'
+        if message == '':
+            message = 'no image needed for this product'
+        for key in form_data:
+            if ('pricecomponent_' in key):
+                # if it is of type "string" the value is the index of the
+                # selection in pricecomponent_name, the id encoded in the key
+                if 'pricecomponent_string' in key:
+                    pricecomponent_id = int(key.split('_')[3])
+                    pricecomponent_value = form_data[key]
+                # if it is of type numerical the value is the actual value of
+                # the pricecomponent field, the id is encoded in the key
+                if 'pricecomponent_int' in key:
+                    pricecomponent_id = int(key.split('_')[3])
+                    pricecomponent_value = form_data[key]
+                # the configurator stores the attributes in a
+                # dictionary in the "attributes" field ,
+                all_attributes[
+                    '_attribute_' + str(pricecomponent_id)
+                    ] = int(pricecomponent_value)
+
+        # add attributes to the configurator
+        chd_dict['attributes'] = str(all_attributes)
+        # dictionary is complete, create configurator
+        new_chd = request.env['chd.product_configurator'].create(chd_dict)
+        # add accessories and price components selections to the configurator
+        for key in form_data:
+            # get only accessories that have been checked
+            if ('accessoryid_' in key) and form_data[key] == 'on':
+                # extrapolate the id encoded the name
+                accessory_id = int(key.split('_')[1])
+                # get the associated value by choosing the field with the
+                # right name accessoryid_{id}=on/off and the associated
+                # quantity would be qtyaccessoryid_{id}=9898
+                accessory_qty = form_data['qty' + key]
+                new_accessory = request.env['chd.accessoire_line'].create({
+                    'product_id': accessory_id,
+                    'configurator_id': new_chd.id,
+                    'quantity': accessory_qty,
+                    })
+                all_accessories.append(new_accessory)
+        # our product configurator is ready, we can now calculate options
+        # _model refers to old API model, self.pool
+        # is not available in controller context (praise the lord for Holger!)
+        try:
+            res = new_chd._model.calculate_price(
+                request.cr, request.uid, [new_chd.id], context=request.context)
+        except orm.except_orm:
+            errormsg = "No result found for the values that you entered. \
+            We would be happy to give you a custom quote. \
+            Please call 010-7856766"
```

Comment:
Can't you get this error message from the error that is raised?
```
except orm.except_orm as e:
    errormsg = e.value
```
Further, wouldn't it be nicer to get the phone number from the company settings? This way, you won't forget changing it if ever the phone number changes